packages/eslint/src/plugins/plugin.ts

```diff
@@ -13,98 +18,152 @@ export interface EslintPluginOptions {
 }
 
 export const createNodes: CreateNodes<EslintPluginOptions> = [
-  combineGlobPatterns(['**/project.json', '**/package.json']),
+  combineGlobPatterns([
+    ...ESLINT_CONFIG_FILENAMES.map((f) => `**/${f}`),
+    baseEsLintConfigFile,
+    baseEsLintFlatConfigFile,
+  ]),
   (configFilePath, options, context) => {
-    const projectRoot = dirname(configFilePath);
-
     options = normalizeOptions(options);
 
-    const eslintConfigs = getEslintConfigsForProject(
-      projectRoot,
-      context.workspaceRoot
-    );
-    if (!eslintConfigs.length) {
-      return {};
+    // TODO: Figure out why this is required to protect against failures in e2e-run
+    (context as any).workspaceFiles = context.workspaceFiles ?? [];
+
+    const configDir = dirname(configFilePath);
+    if (configDir === '.') {
+      return {
+        projects: getProjectsUsingRootESLintConfig(
+          configFilePath,
+          options,
+          context
+        ),
+      };
     }
 
     return {
-      projects: {
-        [projectRoot]: {
-          targets: buildEslintTargets(eslintConfigs, projectRoot, options),
-        },
-      },
+      projects: getProjectsUsingNestedEslintConfig(
+        configFilePath,
+        options,
+        context
+      ),
     };
   },
 ];
 
-function getEslintConfigsForProject(
-  projectRoot: string,
-  workspaceRoot: string
-): string[] {
-  const detectedConfigs = new Set<string>();
-  const baseConfig = findBaseEslintFile(workspaceRoot);
-  if (baseConfig) {
-    detectedConfigs.add(baseConfig);
+function getProjectsUsingRootESLintConfig(
+  configFilePath: string,
+  options: EslintPluginOptions,
+  context: CreateNodesContext
+): CreateNodesResult['projects'] {
+  const projects: CreateNodesResult['projects'] = {};
+
+  // If there's no src folder, it's not a standalone project, so a project.json would be explicitly required to add the root to the mapping
+  const isStandaloneWorkspace =
+    context.workspaceFiles.some((f) => f.startsWith('src')) &&
```

Comment:
This would fail if standalone app's source is not in `src` folder. We need to use smarter condition. The problem is that graph does not exist yet, so we can't check if that folder is a viable project.

@AgentEnder any ideas on how to determine if root is the source of the project?

Reply:
Yeah, we can discuss this for follow up work - this is existing behavior so I'm not going to worry about it in this already large PR.